src/gt4py/next/program_processors/runners/dace_fieldview/transformations/gpu_utils.py

```diff
@@ -0,0 +1,372 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2023, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"""Functions for turning an SDFG into a GPU SDFG."""
+
+import copy
+from typing import Any, Optional, Sequence, Union
+
+import dace
+from dace import properties, transformation
+from dace.sdfg import SDFG, SDFGState, nodes
+
+from gt4py.next.program_processors.runners.dace_fieldview import (
+    transformations as gtx_transformations,
+)
+
+
+__all__ = [
+    "SerialMapPromoterGPU",
+    "GPUSetBlockSize",
+    "gt_gpu_transformation",
+    "gt_set_gpu_blocksize",
+]
+
+
+def gt_gpu_transformation(
+    sdfg: dace.SDFG,
+    try_removing_trivial_maps: bool = True,
+    use_gpu_storage: bool = True,
+    gpu_block_size: Optional[Sequence[int | str] | str] = None,
+    validate: bool = True,
+    validate_all: bool = False,
+    **kwargs: Any,
+) -> dace.SDFG:
+    """Transform an SDFG into an GPU SDFG.
+
+    The transformation expects a rather optimized SDFG and turn it into an SDFG
+    capable of running on the GPU.
+    The function performs the following steps:
+    - If requested, modify the storage location of the non transient arrays to
+        life on GPU.
+    - Call the normal GPU transform function followed by simplify.
+    - If requested try to remove trivial kernels.
+    - If given set the `gpu_block_size` parameters of the Maps to the given value.
+
+    Args:
+        sdfg: The SDFG that should be processed.
+        try_removing_trivial_maps: Try to get rid of trivial maps by incorporating them.
+        use_gpu_storage: Assume that the non global memory is already on the GPU.
```

Comment:
I don't understand this: "the non global memory is already on the GPU". After reading the code, I would change it to:
use_gpu_storage: Set to true when the SDFG array arguments are already allocated on GPU global memory. This will avoid the data copy from host to GPU memory.